src/main/java/com/jwplayer/southpaw/state/RocksDBState.java

```diff
@@ -201,6 +202,14 @@ public void reset() {
      * Rocks DB options
      */
     protected Options rocksDBOptions;
+    /**
+     * RocksDB backup engine
+     */
+    protected BackupEngine backupEngine;
+    /**
+     * Rocks DB backup engine options
+     */
+    protected BackupableDBOptions backupOptions;
```

Comment:
i understand that the current code stores config options within `RocksDBState` but i'd love for this configuration to live within a `RocksDBStateConfig` class where we can validate the configuration before code execution

Reply:
Agreed. Config management is something we need to do better here but out of scope for these changes